### PR TITLE
Add `elim` for Simpl Indexed Type

### LINKED ISSUE #1261: Elim in inductive types
image

### PR DESCRIPTION
**Pull request by HoshinoTented**
Fix #1261

### CODE REVIEW

File: cli-impl/src/test/resources/negative/GoalAndMeta.txt [hunk 1]
```diff
@@ -190,6 +190,15 @@ Error: Unable to solve the type of this literal:
 Let's learn from that.
 
 NonPattern:
+In file C:\Users\hoshino\Documents\Projects\project-aya\aya-dev\cli-impl\src\test\resources\shared\src\data\vec\base.aya:5:6 ->
```
Comment (ice1000) [inline cli-impl/src/test/resources/negative/GoalAndMeta.txt:193]: Why is there absolute path?

**Review by ice1000 — COMMENTED** [review]
Review

File: cli-impl/src/test/resources/negative/GoalAndMeta.txt (hunk 1 above)
Comment (ice1000) [reply cli-impl/src/test/resources/negative/GoalAndMeta.txt:193]: Ah, I think it's related to shadowing warnings.

Comment (HoshinoTented) [reply cli-impl/src/test/resources/negative/GoalAndMeta.txt:193]: omg

**Comment by HoshinoTented** [comment]
no parse error test yet

**Pull request merged**